### PR TITLE
Reimplement list group by username into by user query

### PR DESCRIPTION
**Pull request by wonrax**
## Type

- [x] Feature
- [ ] Bug fix
- [ ] CI/CD
- [ ] Documentation
- [ ] Other

## Description

Brief description of the changes made.

### CODE REVIEW

File: packages/serverless/src/model/global/user.ts [hunk 1]
```diff
@@ -95,7 +95,7 @@ export class ModelUser extends ModelGeneral<OptionalId<TUserRecord>> {
     );
     if (!(await collection.isExist())) {
       await collection.index({ userName: 1 }, { unique: true, session });
-      await collection.index({ publicKey: 1 }, { unique: true, session });
+      await collection.index({ publicKey: 1 }, { unique: false, session });
```
Comment (SangTran-127) [inline packages/serverless/src/model/global/user.ts:98]: I think we should remove unique: false, it's false by default

Comment (wonrax) [reply packages/serverless/src/model/global/user.ts]: fixed

File: packages/serverless/src/model/database/user-group.ts [hunk 2]
```diff
@@ -44,28 +44,40 @@ export class ModelUserGroup extends ModelGeneral<OptionalId<TUserGroupRecord>> {
...
+    const userGroups = this.find({ userName });
     return userGroups.map((userGroup) => userGroup.groupOjectId).toArray();
   }
 
   public async groupNameToGroupId(groupName: string[]): Promise<ObjectId[]> {
     const modelGroup = new ModelGroup(this.databaseName);
-    const availableGroups = await modelGroup.find({
+    const availableGroups = modelGroup.find({
       groupName: { $in: groupName },
     });
```
Comment (SangTran-127) [inline packages/serverless/src/model/database/user-group.ts:82]: availableGroupList, can you double check the find that does find need await or not

Comment (wonrax) [reply packages/serverless/src/model/database/user-group.ts]: fixed

File: packages/serverless/src/apollo/app/group.ts [hunk 3]
```diff
@@ -147,7 +147,9 @@ const groupListByUser = publicWrapper<
   TGroupListByUserRequest,
   TGroupListByUserResponse
 >(JOI_GROUP_LIST_USER, async (_root, { databaseName, userName }) =>
-  new ModelUserGroup(databaseName).listGroupByUserName(userName)
+  (
+    await new ModelUserGroup(databaseName).listGroupByUserQuery({ userName })
```
Comment (chiro-hiro) [inline packages/serverless/src/apollo/app/group.ts:151]: We want the client to able to get group list by user filter, for example the request should be:

```
{
   databaseName: string,
   userQuery: Partial<{
       userName: string,
       publicKey: string,
       email: string
   }>
}
```

Comment (wonrax) [reply packages/serverless/src/apollo/app/group.ts]: fixed

**Review by chiro-hiro — APPROVED** [review]

**Pull request merged**